Add project_configuration_tool option to template

chiroptical opened the pull request:
Closes #20 

- Add an option `project_configuration_tool` which writes `package.yaml` instead of `your-project-name.cabal`
- Added an additional CI step which overrides the default `project_configuration_tool` to hpack to test all configuration options
- I used `stack new example-project` to approximate a minimal `package.yaml` file with the changes you made in the cabal template

chiroptical commented on hooks/post_gen_project.py:
```diff
@@ -0,0 +1,15 @@
+import sys
+
+REMOVE_PATHS = [
+    '{% if cookiecutter.use_hpack == "y" %} {{ cookiecutter.project_name }}.cabal {% endif %}',
+    '{% if cookiecutter.use_hpack != "y" %} package.yaml {% endif %}',
```
Unsure if I should do case insensitive match here

utdemir replied:
I agree that would be better. I think, more importantly, it isn't ideal in the sense that it's not obvious which strings are allowed, and one can enter nonsense which will still pass. Maybe something like [Choice Variables](https://cookiecutter.readthedocs.io/en/latest/advanced/choice_variables.html) functionality might work better there, eg. it could be something like:

```
Select package description format:
1 - Cabal
2 - HPack
Choose from [1], 2:
```

What do you think? It's a bit more wordy, but at least it is obvious that it's a choice between two.

chiroptical replied:
I agree because the default is more obvious. I am just hoping I can make `{{project_name}}.cabal` or `package.yaml (hpack)` or something

chiroptical replied:
Handled in e125757cd58bc0117c0b9dd1fa8ee31f70713f38

chiroptical commented on {{cookiecutter.project_name}}/default.nix:
```diff
@@ -26,6 +26,7 @@ let
       pkgs.haskellPackages.ghcid
       pkgs.haskellPackages.ormolu
       pkgs.haskellPackages.hlint
+      {% if cookiecutter.use_hpack == "y" %}pkgs.haskellPackages.hpack{% endif %}
```
The nix build will pick this up automatically, but it would be nice to have in the shell for local dev

utdemir replied:
I agree, it is definitely good to have. But even after this one needs to remember running `hpack` every time after `package.yaml`. So it would be great if we can also mention this on the README.

chiroptical replied:
Handled in https://github.com/utdemir/hs-nix-template/pull/21/commits/1aeab4d448f93f440e30bc1514a918c2993edf8c

utdemir commented:
Thanks for the PR! I am a bit busy this week, so I will review this over the weekend.

utdemir reviewed (commented):
This is great, I have a few comments, but otherwise happy to merge!

chiroptical commented:
@utdemir I believe I addressed all the issues you pointed out. Thanks for the great input! Also, if you have any other features in mind feel free to bother me about them. I plan to use this template for the foreseeable future and would be happy to dev on it.
